src/components/course/SubsidyRequestButton.jsx

```diff
@@ -31,16 +31,14 @@ const SubsidyRequestButton = () => {
     refreshSubsidyRequests,
   } = useContext(SubsidyRequestsContext);
 
-  const { state, subsidyRequestCatalogsApplicableToCourse } = useContext(CourseContext);
+  const { state, subsidyRequestCatalogsApplicableToCourse, userSubsidyApplicableToCourse } = useContext(CourseContext);
```

Comment:
`userSubsidyApplicableToCourse` moved outside of `state` where it was previously housed. The stuff in `state` gets passed into a `useReducer` which means you have to use reducer actions to mutate the data, which isn't great for `userSubsidyApplicableToCourse` which may update at any time.

This fix pulls `userSubsidyApplicableToCourse` from its new location instead of from `state`.